[#181854197] remove elasticsearch

whi-tw opened the pull request:
## What

Describe what you have changed and why.

## How to review

Describe the steps required to test the changes.

## Who can review

Describe who can review the changes. Or more importantly, list the people
that can't review, because they worked on it.

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

whi-tw committed:
Remove ElasticSearch code

As ElasticSearch has now been completely eliminated from Aiven,
we need to remove the legacy code from the broker.

AP-Hunt commented on provider/provider_test.go:
```diff
@@ -35,11 +35,11 @@ var _ = Describe("Provider", func() {
 		planSpecificConfig1 := provider.PlanSpecificConfig{}
 		planSpecificConfig1.AivenPlan = "startup-1"
-		planSpecificConfig1.ElasticsearchVersion = "6"
+		planSpecificConfig1.OpenSearchVersion = "6"
```
We ought to correct this to the real major version of OpenSearch: 1.

AP-Hunt committed:
Removes Elasticsearch reference from example config

AP-Hunt committed:
Replace link to `provider/config.go` with a more up to date one

The previous link was to an old commit. The new one points to the tip of the
`main` branch, so ought to stay more up-to-date.
